worlds/witness/hints.py

```diff
@@ -583,12 +602,28 @@ def create_all_hints(world: "WitnessWorld", hint_amount: int, area_hints: int,
 def make_compact_hint_data(hint: WitnessWordedHint, local_player_number: int) -> CompactItemData:
     location = hint.location
     area_amount = hint.area_amount
-
-    # None if junk hint, address if location hint, area string if area hint
-    arg_1 = location.address if location else (hint.area if hint.area else None)
-
-    # self.player if junk hint, player if location hint, progression amount if area hint
-    arg_2 = area_amount if area_amount is not None else (location.player if location else local_player_number)
+    hunt_panels = hint.area_hunt_panels
+
+    # -1 if junk hint, address if location hint, area string if area hint
+    arg_1: Union[str, int]
```

Comment:
I'm not a big fan of these vague and generic variable names. I understand they can hold multiple different things, but here's what I would do

Turn the whole first arg code into a different method, where the arg_1 is actually multiple variables, or even better, no variable, with just some "return [whatever]" at the relevant lines. Then do the same for the arg_2 code.
Then, here, you can also not use the variables at all, and just return the method calls

But that's definitely not critical, feel free to ignore